_use_cases/schema_validation_and_enforecement.markdown

```diff
@@ -0,0 +1,84 @@
+---
+name: Schema Validation and Enforcement
+---
+
+### Schema Validation and Enforcement
+
+#### Why
+
+In Kafka, producing applications transfer messages to consuming applications.  In order for messages to be transferred successfully
+producer and consumer need to agree on the format of the message being transferred. If there is a mismatch between the format sent
+by the producer and the expectations of the consumer, problems will result.  The consuming application could take the wrong business
+action or the consuming application may fail leading to complete failure of the system.  This problem is known as *poison message*
+scenario.
+
+If there is only a small number of applications, perhaps all managed by the same team, it is possible for the developers to use
+informal agreements about what message format will be used on which topic.  However, as the use of Kafka grows within the organization,
+the amount of effort required to keep all the formats in agreement grows.  Mistakes, and system downtime, become inevitably.
+
+| ![image]({{'/assets/pages/images/schema_validation_problem.png' | absolute_url}}){:width="100%"} |
+|:-----------------------------------------------------------------:|
+|    *Problem: Poison message leading to consumer crash*     |
+
+
+##### Schemas to the rescue
+
+To overcome the problem, the Kafka **client** ecosystem supports _schemas_.  Schemas provide a programmatic description
+of the message format.
+
+Producers use a schema to ensure that they produce messages that adhere to the agreed message format.  The consumer
```

Comment:
My mental model was always
> producers use a schema to advertise the format of the messages they (are supposed to) produce

Which is where a proxy would step in to ensure the compliance with the schema.